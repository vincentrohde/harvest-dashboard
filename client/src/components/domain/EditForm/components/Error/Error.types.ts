type errorList = string[];

export interface ErrorProps {
    error: errorList;
}
