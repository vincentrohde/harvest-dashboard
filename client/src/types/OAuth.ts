export type oAuthType = {
    access_token: string;
    account_id: string;
    expiration_date: string;
    refresh_token: string;
};
