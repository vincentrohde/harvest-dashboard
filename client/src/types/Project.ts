import { categoriesType, CategoryInterface } from './Category';

export type projectsType = categoriesType;

export interface ProjectInterface extends CategoryInterface {}

export type projectsByHours = ProjectByHours[];

export interface ProjectByHours {
    project: string;
    hours: number;
}
