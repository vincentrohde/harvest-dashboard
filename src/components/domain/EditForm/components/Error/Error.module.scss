.Error {
    margin: 0 0 1em;
}
