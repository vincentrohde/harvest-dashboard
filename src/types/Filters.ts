export interface FiltersInterface {
    dateRange: string[] | 1;
}
