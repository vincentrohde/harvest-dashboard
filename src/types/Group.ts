export type groupsType = GroupInterface[];

export interface GroupInterface {
    id: number;
    name: string;
}