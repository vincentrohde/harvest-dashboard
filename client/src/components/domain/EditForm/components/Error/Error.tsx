// Libs
import React from 'react';
import { Message } from "semantic-ui-react";

// Types
import { ErrorProps } from './Error.types';

// Styling
import './Error.scss';

const Error = ({ error }: ErrorProps) => {
    let errorContent: string;
    switch (error[0]) {
        case 'hours':
            errorContent = 'Hours need to be in format (H)H:MM';
            break;
        case 'spent_date':
            errorContent = 'Date needs to be in format DD.MM.YYYY';
            break;
        default:
            errorContent = 'There is something wrong with your inputs';
    }
    return (
        <div className="Error">
            { error && (
                <Message
                    error
                    header="Input Error"
                    content={ errorContent }
                />
            )}
        </div>
    );
}

export default Error;
