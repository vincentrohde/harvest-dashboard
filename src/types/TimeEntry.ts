import { ProjectInterface } from './Project';
import { TaskInterface } from './Task';

export type timeEntriesType = TimeEntryInterface[];

export interface ShallowTimeEntryInterface {
    hours: number | string;
    id: number;
    notes: string;
    spent_date: string;
    is_running: boolean;
    project: ProjectInterface;
    task: TaskInterface;
}

export interface TimeEntryInterface extends ShallowTimeEntryInterface {
    hours: number;
}

export interface ShallowSubmissionEntryInterface {
    notes: ShallowTimeEntryInterface['notes'];
    spent_date: ShallowTimeEntryInterface['spent_date'];
    hours: ShallowTimeEntryInterface['hours'];
    project_id: number | string;
    task_id: number | string;
}

export interface TimeEntrySubmissionInterface extends ShallowSubmissionEntryInterface {
    hours: TimeEntryInterface['hours'];
    project_id: number;
    task_id: number;
}