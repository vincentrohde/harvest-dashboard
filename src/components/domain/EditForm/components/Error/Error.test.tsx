// Libs
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

// Components
import Error from './Error';

// Types
import { ErrorProps } from './Error.types';

describe('<TabContainer />', () => {
    it('renders hours error correctly', () => {
        const mockData: ErrorProps = {
            error: ['hours']
        }

        const { asFragment } = render(
            <Error {...mockData} />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('renders spent_date error correctly', () => {
        const mockData: ErrorProps = {
            error: ['spent_date']
        }

        const { asFragment } = render(
            <Error {...mockData} />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('renders default error correctly', () => {
        const mockData: ErrorProps = {
            error: []
        }

        const { asFragment } = render(
            <Error {...mockData} />
        );

        expect(asFragment()).toMatchSnapshot();
    });
})
