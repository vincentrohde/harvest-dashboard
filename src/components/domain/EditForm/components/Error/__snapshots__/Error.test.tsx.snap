// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabContainer /> renders default error correctly 1`] = `
<DocumentFragment>
  <div
    class="Error"
  >
    <div
      class="ui error message"
    >
      <div
        class="content"
      >
        <div
          class="header"
        >
          Input Error
        </div>
        <p>
          There is something wrong with your inputs
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<TabContainer /> renders hours error correctly 1`] = `
<DocumentFragment>
  <div
    class="Error"
  >
    <div
      class="ui error message"
    >
      <div
        class="content"
      >
        <div
          class="header"
        >
          Input Error
        </div>
        <p>
          Hours need to be in format (H)H:MM
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<TabContainer /> renders spent_date error correctly 1`] = `
<DocumentFragment>
  <div
    class="Error"
  >
    <div
      class="ui error message"
    >
      <div
        class="content"
      >
        <div
          class="header"
        >
          Input Error
        </div>
        <p>
          Date needs to be in format DD.MM.YYYY
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
